dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/event/EventOperationParamsMapper.java

```diff
@@ -0,0 +1,321 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker.export.event;
+
+import static org.apache.commons.lang3.BooleanUtils.toBooleanDefaultIfNull;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.hisp.dhis.category.CategoryOptionCombo;
+import org.hisp.dhis.common.OrganisationUnitSelectionMode;
+import org.hisp.dhis.feedback.BadRequestException;
+import org.hisp.dhis.feedback.ForbiddenException;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.organisationunit.OrganisationUnitService;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramService;
+import org.hisp.dhis.program.ProgramStage;
+import org.hisp.dhis.program.ProgramStageService;
+import org.hisp.dhis.security.Authorities;
+import org.hisp.dhis.security.acl.AclService;
+import org.hisp.dhis.trackedentity.TrackedEntity;
+import org.hisp.dhis.trackedentity.TrackedEntityService;
+import org.hisp.dhis.trackedentity.TrackerAccessManager;
+import org.hisp.dhis.user.CurrentUserService;
+import org.hisp.dhis.user.User;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Maps {@link EventOperationParams} to {@link EventSearchParams} which is used
+ * to fetch enrollments from the DB.
+ */
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class EventOperationParamsMapper
+{
+
+    private final ProgramService programService;
+
+    private final ProgramStageService programStageService;
+
+    private final OrganisationUnitService organisationUnitService;
+
+    private final TrackedEntityService trackedEntityService;
+
+    private final AclService aclService;
+
+    private final CategoryOptionComboService categoryOptionComboService;
+
+    private final TrackerAccessManager trackerAccessManager;
+
+    private final CurrentUserService currentUserService;
+
+    //For now this maps to EventSearchParams. We should create a new EventQueryParams class that should be used in the persistence layer
+    @Transactional( readOnly = true )
+    public EventSearchParams map( EventOperationParams operationParams )
+        throws BadRequestException,
+        ForbiddenException
+    {
+        User user = currentUserService.getCurrentUser();
+
+        Program program = validateProgram( operationParams.getProgramUid() );
```

Comment:
I have missed this in reviewing @enricocolasante changes to the enrollment controller. By moving these validations from the service into the mapper we cannot use them when validating operations for single entities. `/tracker/trackedEntites/{te}?program={uid}` is such a case.

Reply:
I can't find where right now, but I remember reading somewhere that single entities requests are out of the scope of this refactor, are they not?